app/src/main/java/pl/droidevs/books/library/LibraryActivity.java

```diff
@@ -108,60 +99,78 @@ protected void onCreate(Bundle savedInstanceState) {
-            ImageView imageView = view.findViewById(R.id.iv_book);
-            TextView titleTextView = view.findViewById(R.id.tv_book_title);
-            TextView authorTextView = view.findViewById(R.id.tv_book_author);
-            View shadowView = view.findViewById(R.id.shadow_view);
-
-            Intent intent = new Intent(this, BookActivity.class);
-            Bundle extras = new Bundle();
-            extras.putString(EXTRAS_IMAGE_TRANSITION_NAME, imageView.getTransitionName());
-            extras.putString(EXTRAS_TITLE_TRANSITION_NAME, titleTextView.getTransitionName());
-            extras.putString(EXTRAS_AUTHOR_TRANSITION_NAME, authorTextView.getTransitionName());
-            extras.putString(EXTRAS_SHADOW_TRANSITION_NAME, shadowView.getTransitionName());
-            extras.putInt(EXTRAS_LAST_SELECTED_INDEX, index);
-            intent.putExtra(BUNDLE_EXTRAS, extras);
-            intent.putExtra(EXTRAS_BOOK_ID, bookId);
-
-            ActivityOptionsCompat options = ActivityOptionsCompat.makeSceneTransitionAnimation(
-                    this,
-                    new Pair<>(imageView, imageView.getTransitionName()),
-                    new Pair<>(titleTextView, titleTextView.getTransitionName()),
-                    new Pair<>(authorTextView, authorTextView.getTransitionName()),
-                    new Pair<>(shadowView, shadowView.getTransitionName())
-            );
-
-            ActivityCompat.startActivityForResult(this, intent, BOOK_REQUEST, options.toBundle());
+
+            ActivityCompat.startActivityForResult(this,
+                    createBookIntent(view, index, bookId),
+                    BOOK_REQUEST,
+                    createBookActivityOptions(view).toBundle());
         });
     }
 
+    private Intent createBookIntent(View view, int index, BookId bookId) {
+        Intent intent = new Intent(this, BookActivity.class);
+        intent.putExtra(BUNDLE_EXTRAS, createAnimationBundle(view, index));
+        intent.putExtra(EXTRAS_BOOK_ID, bookId);
+
+        return intent;
+    }
+
+    private Bundle createAnimationBundle(View view, int index) {
+        Bundle animationBundle = new Bundle();
+        animationBundle.putString(EXTRAS_IMAGE_TRANSITION_NAME, view.findViewById(R.id.iv_book).getTransitionName());
+        animationBundle.putString(EXTRAS_TITLE_TRANSITION_NAME, view.findViewById(R.id.tv_book_title).getTransitionName());
+        animationBundle.putString(EXTRAS_AUTHOR_TRANSITION_NAME, view.findViewById(R.id.tv_book_author).getTransitionName());
+        animationBundle.putString(EXTRAS_SHADOW_TRANSITION_NAME, view.findViewById(R.id.shadow_view).getTransitionName());
+        animationBundle.putInt(EXTRAS_LAST_SELECTED_INDEX, index);
+
+        return animationBundle;
+    }
+
+    private ActivityOptionsCompat createBookActivityOptions(View view) {
+        return ActivityOptionsCompat.makeSceneTransitionAnimation(
+                this,
```

Comment:
To, co można by ewentualnie zrobić to stworzyć jakąś klasę helpera która by ustawiała te bundle, żeby nie trzymać tego w activity. Ale to może być potem zrobione moim zdaniem :)